crates/fiber/src/unix/aarch64.rs

```diff
@@ -0,0 +1,177 @@
+
+        // Switch to the new stack and restore all our callee-saved
+        // registers after the switch and return to our new stack.
+        mov sp, x8
+        ldp d15, d14, [sp], 16
+        ldp d13, d12, [sp], 16
+        ldp d11, d10, [sp], 16
+        ldp d9, d8, [sp], 16
+        ldp x28, x27, [sp], 16
+        ldp x26, x25, [sp], 16
+        ldp x24, x23, [sp], 16
+        ldp x22, x21, [sp], 16
+        ldp x20, x19, [sp], 16
+        ldp x29, x30, [sp], 16
+    ",
+    autiasp!(),
+    cfi_window_save!(),
+    "
+        ret
+        .cfi_endproc
+    ",
+);
+
+// fn(
+//    top_of_stack(%x0): *mut u8,
+//    entry_point(%x1): extern fn(*mut u8, *mut u8),
+//    entry_arg0(%x2): *mut u8,
+// )
+// We set up the newly initialized fiber, so that it resumes execution
+// from wasmtime_fiber_start(). As a result, we need a signed address
+// of this function, so there are 2 requirements:
+// * The fiber stack pointer value that is used by the signing operation
+//   must match the value when the pointer is authenticated inside
+//   wasmtime_fiber_switch(), otherwise the latter would fault
+// * We would like to use an instruction that is executed as a no-op by
+//   processors that do not support PAuth, so that the code is
+//   backward- compatible and there is no duplication; `PACIA1716` is a
+//   suitable one, which has the following operand register
+//   conventions:
+//   * X17 contains the pointer value to sign
+//   * X16 contains the modifier value
+//
+// TODO: Use the PACGA instruction to authenticate the saved register
+// state, which avoids creating signed pointers to
+// wasmtime_fiber_start(), and provides wider coverage.
+#[rustfmt::skip]
+asm_func!(
+    "wasmtime_fiber_init",
+    "
+        .cfi_startproc
+        hint #34 // bti c
+        sub x16, x0, #16
+        adr x17, ", asm_sym!("wasmtime_fiber_start"), "
+    ",
+    pacia1716!(),
+    "
+        str x17, [x16, -0x8] // x17 => lr
+        str x0, [x16, -0x18] // x0 => x19
+        stp x2, x1, [x0, -0x38] // x1 => x20, x2 => x21
+
+        // `wasmtime_fiber_switch` has an 0xa0 byte stack, and we add 0x10 more for
+        // the original reserved 16 bytes.
+        add x8, x0, -0xb0
+        str x8, [x0, -0x10]
+        ret
+        .cfi_endproc
+    ",
+);
+
+// See the x86_64 file for more commentary on what these CFI directives are
+// doing. Like over there note that the relative offsets to registers here
+// match the frame layout in `wasmtime_fiber_switch`.
+asm_func!(
+    "wasmtime_fiber_start",
+    "
+        .cfi_startproc simple
+        .cfi_def_cfa_offset 0
+        .cfi_escape 0x0f,    /* DW_CFA_def_cfa_expression */ \
+            5,               /* the byte length of this expression */ \
+            0x6f,            /* DW_OP_reg31(%sp) */ \
+            0x06,            /* DW_OP_deref */ \
+            0x23, 0xa0, 0x1  /* DW_OP_plus_uconst 0xa0 */
+        .cfi_rel_offset x29, -0x10
+    ",
+    cfi_window_save!(),
```

Comment:
Nit - do you mind moving this after the next directive (about x30), since it refers to the same register? It would look a bit nicer IMHO.